joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/ast/AstForStatementsCreator.scala

```diff
@@ -341,15 +341,21 @@ trait AstForStatementsCreator(implicit withSchemaValidation: ValidationMode) {
   def astForWhenAsStatement(expr: KtWhenExpression, argIdx: Option[Int])(implicit
     typeInfoProvider: TypeInfoProvider
   ): Ast = {
-    assert(expr.getSubjectExpression != null)
-
-    val astForSubject = astsForExpression(expr.getSubjectExpression, Some(1)).headOption.getOrElse(Ast())
-    val finalAstForSubject = expr.getSubjectExpression match {
-      case p: KtProperty =>
-        val block = blockNode(p, "", "").argumentIndex(1)
-        blockAst(block, List(astForSubject))
-      case _ => astForSubject
+    val (astForSubject, finalAstForSubject) = Some(expr.getSubjectExpression) match {
```

Comment:
`Some(null)` will return `Some(null)` and not `None`
```suggestion
    val (astForSubject, finalAstForSubject) = Option(expr.getSubjectExpression) match {
```

Reply:
@DavidBakerEffendi thnx for catching this one. I will make changes and update.

Reply:
Done